memgpt/schemas/document.py

```diff
@@ -0,0 +1,14 @@
+import uuid
+from typing import Dict, Optional
+
+from pydantic import BaseModel, Field
+
+
+class DocumentModel(BaseModel):
```

Comment:
now for the hard part :) 
if you take a look at the [Document ORM](https://github.com/cpacker/MemGPT/blob/1.0.0-pre/memgpt/orm/document.py) it has relationships for the `Organization` it belongs to, and the `Passages` that belong to it. When you build the `Document` object you'll probably want some of the info from those relationships, but not all of it. If you did the whole thing like this 

```python
class DocumentModel(BaseModel):
... 
organization: Organization
passages: List[Passage] 
```
you get infinite recursion (because `Organization` has `documents` which is a list including this object) and also an obese object full of stuff you don't need. 

so maybe you have a schema called `OrganizationSummary` that only includes the id and the name of the org, maybe the org's email domain? those seem useful without getting too verbose. Then it looks like this:

```python
class DocumentModel(BaseModel):
... 
organization: OrganizationSummary
passages: List[PassageSummary] 

>> doc = DocumentModel(**some_vals)
>> doc.organization.name
'Some Great Company'

>> doc.organization.email_domain
'somegreatcompany.org'

>> doc.organization.users
AttributeError: OrganizationSummary has no attribute 'users'

# just for good measure, get users from the form
>> sql_org = doc.organization.to_sqlalchemy(db_session)
>> sql_org.users
[<User 'Dave'>, <User 'Lucy'>, <User 'Saresh'>]

```
We don't need to get this perfect right away, but the idea is to balance giving the object enough useful information that you need all the time to avoid tons of trips back to the db, vs making the objects obese and adding relationship complexity you don't care about.

Reply:
I think we should keep all the schemas as trim as possible for now, and maybe add additional stuff as needed when we update `server.py`